RocketChat/Rocket.Chat
# feat: Replace "Hide Usernames" by the new "Messages Layout" preference

## Proposed changes (including videos or screenshots)


 - Replace "Hide Usernames" preference and "Use Real Name" setting by the **new "Messages Layout" preference** (with the "Show only username", "Show only full name" and "Show both full names and username" options);



## Issue(s)


## Steps to test or reproduce

Login and go to **My Account > Preferences > Messages** and choose one of the three options provided. Messages lists should be updated accordingly (threads' and users' lists should also honor the preference's value).
The default value for this setting can be chosen in **Workspace > Settings > Accounts > Default User Preferences > Default Messages Layout**.

## Further comments


The new setting's inital value will be assigned according to the "Use Real Name" setting' value AND the "Hide Usernames" preference's default value:

| Use Real Name | Default "Hide Usernames" | Default Messages Layout |
|---|---|---|
| true | true | Show only full name |
| true | false | Show both full name and username |
| false | false | Show only username |

TC-157
